cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ScheduleSpecificationCodec.java

```diff
@@ -32,11 +34,48 @@
  *
  */
 public class ScheduleSpecificationCodec extends AbstractSpecificationCodec<ScheduleSpecification> {
+
+  /**
+   * Schedule Type.
+   */
+  private enum ScheduleType {
+    ORIGINAL_TIME,
```

Comment:
Seems like we should just deprecate the `Schedule` class and replace it with the `TimeSchedule` class? We can create `TimeSchedule` class if someone provides the `Schedule` class through the configurer such that internally there is no `Schedule`.